### DISCUSSION TITLE
Translation

### DISCUSSION THREAD

**Discussion by SonnyBrotherman** [owncast/owncast#720] [Ideas]
I personally had to change translations in source files to have everything in Swedish..

From Admin Dashboard
- Possibility to translate in simple fields, string translation.

Let me know if you need translation. I'm ready for Swedish translation if needed.

**Comment by gabek** [comment]
I think if we had a string-heavier interface I could understand the need for some localization.  But realistically there's only a few strings in the UI, and (in my opinion) none of them are really that important.  Most of the UI is built via customized content, and both the video player and chat doesn't really require any localized understanding to use it.

I'm certainly not opposed, but having an entire localization framework for four or five unimportant strings seems like a lot for little benefit for most people.

What strings would you put on the list of needing translating?

**Reply by SonnyBrotherman** [reply]
No, you're right.

Perhaps what I'm looking for is which files to open and change for a translation in the Owncast Docs.
I had to go through every single file to figure it out.

For example:

```
To translate the front-end these are the files you should edit.
x
y
z
```

**Reply by jeyemwey** [reply]
Before we maintain another list, far away from the code itself, I'd vote for a simple small language system. Not a large solution, but a common place to save text in the frontend.

**Reply by gingervitis** [reply]
I could see a relatively straight forward, but very manual, system where we had a local JSON file of all strings of the app, then populated the app with it.  It'd be yet another file or request that the app would have to load in though.
```
// available translatable strings

{
  streamOnline: 'Stream is online.',
  streamOffline: 'Offline.',
  socialHandleHeader: 'FOLLOW ME',

  chatInitialPrompt: 'Type here to chat, no account neccesary',
  chatPrompt: 'Message',
  ...
}
```

**Reply by aaronpk** [reply]
I would highly recommend looking into an actual localization framework instead of coming up with something from scratch if you want to go down that route. They'll have accounted for edge cases you may not have thought about.

**Reply by gabek** [reply]
Since this came up today and I never responded last time:

I totally agree that if it's to be done it needs to be using an actual localization framework.  Things like concatenation, pluralization, etc when it comes to localized strings are more than just edge cases, they're regular use of language that are so easy to get wrong without having the structure around it.

**Comment by gabek** [comment]
Just wanted to close this with a link to the translation work that is in progress.
